File: src/ros_projects/src/server_client/py/area_server.py
#!/usr/bin/env python3
import rospy
from ros_projects.srv import RectangleArea
from ros_projects.srv import RectangleAreaRequest
from ros_projects.srv import RectangleAreaResponse
import time

def calculate_area(req):
    print("Area of width: %s height: %s is %s " %(req.width, req.height, (req.width * req.height)))
    time.sleep(0.2)
    response = RectangleAreaResponse(req.width * req.height)
    return response

if __name__ == "__main__":
    # Init node
    rospy.init_node("calculate_area_server")
    # Init service as calculate_area
    service = rospy.Service("calculate_area", RectangleArea, calculate_area)
    print("Ready to calculate the area")
    # Infinite loop
    rospy.spin()

File: ros_projects/src/server_client/cpp/area_server.cpp
#include "ros/ros.h"
#include "ros_projects/RectangleArea.h"

bool calculate(ros_projects::RectangleArea::Request  &req,
         ros_projects::RectangleArea::Response &res)
{
  res.area = req.width * req.height;
  ROS_INFO("request: x=%ld, y=%ld", (long int)req.width, (long int)req.height);
  ROS_INFO("sending back response: [%ld]", (long int)res.area);
  return true;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "rectangle_area_server");
  ros::NodeHandle server_node;

  ros::ServiceServer service = server_node.advertiseService("rectangle_area", calculate);
  ROS_INFO("Ready to calculate the area");
  ros::spin();

  return 0;
}
